tests/Integration/Infrastructure/Controller/ListRegulationsControllerTest.php

```diff
@@ -11,35 +11,43 @@ final class ListRegulationsControllerTest extends WebTestCase
     public function testList(): void
     {
         $client = static::createClient();
-        $crawler = $client->request('GET', '/');
+        $pageOne = $client->request('GET', '/?pageSize=1');
 
         $this->assertResponseStatusCodeSame(200);
-        $this->assertSame('Réglementations', $crawler->filter('h3')->text());
+        $this->assertSame('Réglementations', $pageOne->filter('h3')->text());
 
-        $tabs = $crawler->filter('.fr-tabs__list')->eq(0);
+        $tabs = $pageOne->filter('.fr-tabs__list')->eq(0);
 
         $this->assertSame("tablist", $tabs->attr("role"));
         $this->assertSame("Brouillons (2) Publiée (1)", $tabs->text());
 
-        $draftRows = $crawler->filter('#draft-panel tbody > tr');
-        $this->assertSame(2, $draftRows->count());
+        $pageOneDraftRows = $pageOne->filter('#draft-panel tbody > tr');
+        $this->assertSame(1, $pageOneDraftRows->count()); // One item per page
 
-        $draftRow0 = $draftRows->eq(0)->filter('td');
-        $this->assertEmpty($draftRow0->eq(0)->text()); // No location
-        $this->assertEmpty($draftRow0->eq(1)->text()); // No period set
-        $this->assertSame("Brouillon", $draftRow0->eq(2)->text());
+        $pageOneDraftRow0 = $pageOneDraftRows->eq(0)->filter('td');
+        $this->assertEmpty($pageOneDraftRow0->eq(0)->text()); // No location
+        $this->assertEmpty($pageOneDraftRow0->eq(1)->text()); // No period set
+        $this->assertSame("Brouillon", $pageOneDraftRow0->eq(2)->text());
 
-        $draftRow1 = $draftRows->eq(1)->filter('td');
-        $this->assertSame("Savenay Route du Grand Brossais", $draftRow1->eq(0)->text());
-        $this->assertSame("du 08/12/2022 au 18/12/2022", $draftRow1->eq(1)->text());
-        $this->assertSame("Brouillon", $draftRow1->eq(2)->text());
+        $pageOnePublishedRows = $pageOne->filter('#published-panel tbody > tr');
+        $this->assertSame(1, $pageOnePublishedRows->count());
 
-        $publishedRows = $crawler->filter('#published-panel tbody > tr');
-        $this->assertSame(1, $publishedRows->count());
+        $pageOnePublishedRow0 = $pageOnePublishedRows->eq(0)->filter('td');
+        $this->assertEmpty($pageOnePublishedRow0->eq(0)->text()); // No location
+        $this->assertSame("depuis le 08/10/2022 permanent", $pageOnePublishedRow0->eq(1)->text());
+        $this->assertSame("Réglementation en cours", $pageOnePublishedRow0->eq(2)->text());
 
-        $publishedRow0 = $publishedRows->eq(0)->filter('td');
-        $this->assertEmpty($publishedRow0->eq(0)->text()); // No location
-        $this->assertSame("depuis le 08/10/2022 permanent", $publishedRow0->eq(1)->text());
-        $this->assertSame("Réglementation en cours", $publishedRow0->eq(2)->text());
+        $pageTwo = $client->request('GET', '/2?activeTab=draft&pageSize=1');
```

Comment:
Je viens de comprendre grâce à cette ligne que la page est encodée dans un paramètre de chemin, `/{page}`

Ça me surprend un peu, je pensais que ce serait exclusivement un query parameter : `?page=...`

J'allais dès lors faire la suggestion suivante

```suggestion
        $pageTwo = $client->request('GET', '/?page=2&pageSize=1');
```

Je crois que `activeTab` n'est pas nécessaire ici, puisque par défaut on affiche les brouillons sur la page d'accueil, et d'ailleurs on l'omet dans la première requête.

Par contre il faudrait faire un test sur le `activeTab=published` ? Cela nécessite une fixture d'une réglementation publiée, si je comprends bien.

Reply:
Je ne sais pas si c'est nécessaire de créer un test supplémentaire spécifique, car l'onglet publié est également testé. Je vérifie dans le DOM la présence de la réglementation.